x/observer/keeper/ballot_test.go

```diff
@@ -4,20 +4,92 @@ import (
 	"testing"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
 	"github.com/zeta-chain/zetacore/x/observer/types"
 )
 
 func TestKeeper_GetBallot(t *testing.T) {
 	k, ctx := SetupKeeper(t)
 	identifier := "0x9ea007f0f60e32d58577a8cf25678942d2b10791c2a34f48e237b76a7e998e4d"
-	k.SetBallot(ctx, &types.Ballot{
-		Index:            "",
-		BallotIdentifier: identifier,
-		VoterList:        nil,
-		ObservationType:  0,
-		BallotThreshold:  sdk.Dec{},
-		BallotStatus:     0,
-	})
-
-	k.GetBallot(ctx, identifier)
+	b := &types.Ballot{
+		Index:                "",
+		BallotIdentifier:     identifier,
+		VoterList:            nil,
+		ObservationType:      0,
+		BallotThreshold:      sdk.Dec{},
+		BallotStatus:         0,
+		BallotCreationHeight: 1,
+	}
+	_, found := k.GetBallot(ctx, identifier)
+	require.False(t, found)
+
+	k.SetBallot(ctx, b)
+
+	ballot, found := k.GetBallot(ctx, identifier)
+	require.True(t, found)
+	require.Equal(t, *b, ballot)
+}
+
+func TestKeeper_GetBallotList(t *testing.T) {
+	k, ctx := SetupKeeper(t)
+	identifier := "0x9ea007f0f60e32d58577a8cf25678942d2b10791c2a34f48e237b76a7e998e4d"
```

Comment:
We can also use `sample.ZetaIndex(t)` the ballot identifier is the same as the cctx index for the inbound ballots, 
or you could create a sample function to generate a random index as well